File: 10/main.cpp
#include <iostream>
#include <map>
#include <string>

// решение на C++ по каким-то причинам не принимается, но
// идентичное на Go проходит без проблем.

int main()
{
	std::string s;
	std::cin >> s;
	std::map<char, int> counts;
	int n = s.length();
	int left = 0;
	int right = n - 1;
	while (left <= right)
	{
		int count = (n - left) * (left + 1);
		counts[s[left]] += count;
		if (left != right)
			counts[s[right]] += count;
		left++;
		right--;
	}
	for (const auto& p : counts)
		std::cout << p.first << ':' << ' ' << p.second << std::endl;
	return 0;
}

File: 10/main.go
package main

import "fmt"

const SIZE int = 26

func main() {
	var s string
	fmt.Scan(&s)
	var counts [SIZE]int
	for i := 0; i < SIZE; i++ {
		counts[i] = 0
	}
	n := len(s)
	left := 0
	right := n - 1
	for left <= right {
		count := (n - left) * (left + 1)
		counts[s[left] - 'a'] += count
		if left != right {
			counts[s[right] - 'a'] += count
		}
		left++;
		right--;
	}
	for i := 0; i < SIZE; i++ {
		if (counts[i] != 0) {
			fmt.Printf("%c: %d\n", i + 'a', counts[i])
		}
	}
}
